### PR TITLE
Improved `if` and `function` Javascript snippets.

### PR DESCRIPTION
**Pull request by mrmartineau**
- removed unneeded semicolons after them
- improved whitespace

It always bugged me that there were needless semicolons after these snippets..

### CODE REVIEW

File: JavaScript/function.sublime-snippet [hunk 1]
```diff
@@ -1,5 +1,7 @@
 <snippet>
-    <content><![CDATA[function($1) {${0:$TM_SELECTED_TEXT}};]]></content>
+    <content><![CDATA[function($1) {
+	${0:$TM_SELECTED_TEXT}
+}]]></content>
```
Comment (FichteFoll) [inline JavaScript/function.sublime-snippet]: I don't think it's a good idea to always make this multi-line. You can easily turn the single-line anonymous function into a multi-line one by pressing enter/return once, but not the other way around.

**Comment by FichteFoll** [comment]
Closes https://github.com/SublimeTextIssues/DefaultPackages/issues/37.

**Comment by sublimehq** [comment]
I agree with @FichteFoll here: the changes in general are great, but adding the linebreaks is likely a step in the wrong direction

**Comment by mrmartineau** [comment]
Your points are valid, how about I resubmit without the line breaks?

**Comment by FichteFoll** [comment]
I'd like to mention again that the for snippets are missing in this pull request and still have semicolons.

**Comment by mrmartineau** [comment]
@FichteFoll I didn't notice your edit about the `for` snippets & I didn't notice that they had semicolons as well.. I will submit another PR later today

**Comment by mrmartineau** [comment]
removed the semicolons for `for` loops now :)

**Comment by poke** [comment]
You should squash those changes and repush them (`--force`) so that the intermediary states are not included.

Also, the change in `function-(fun).sublime-snippet` is introducing a broken syntax as there is a line comment. The snippet should stay multi-line as it currently is. While I would personally welcome multi-line syntax for all the snippets, I accept that the consensus seems to be that they should stay one-line ones. But the ones that are currently multi-line shouldn’t be changed to one-liners.

Finally, is there a reason for the changed placeholder order in `if-___-else.sublime-snippet`? It seems a bit weird to first insert the else part…

**Comment by mrmartineau** [comment]
I have created a new, better PR #43 which should be used instead. Thanks for your input, it is nice to know you all care about the details.

**Pull request closed without merging**